### ISSUE TITLE
Small feature request: Add llvm as git submodule 

### ISSUE THREAD

**Issue by maxwillzq** [openxla/stablehlo#1021]
### Request description

Currently during the "build step" of readme (https://github.com/openxla/stablehlo/blob/main/README.md), 
it ask users to handle llvm download and version control.

We should use git submodule to handle it better. (https://www.geeksforgeeks.org/submodules-in-git/)
Then we can check out the specific commit of llvm repo we need.

**Comment by GleasonK** [comment]
Hello! Thanks for opening this ticket.

We had explored this in the past in https://github.com/openxla/stablehlo/pull/27 and the resolution was to avoid using submodules until we had a great use case for it. I gathered feedback from similar projects and it seems many have hit issues with submodules. Another driving force was that many users get StableHLO via mlir-hlo to be able to use optimizations/machinery in that repo, so it is likely best to keep our build in line with the mlir-hlo build, which uses a similar `llvm_version.txt` setup.

Let me know if your use case has a strong reason for preferring a submodule over the bring-your-own LLVM approach! Happy to re-open that discussion as needed. Otherwise, we can use this ticket to track other requests for submodules, can close it for now and re-open as the demand for submodules increases.
